src/phases.rs

```diff
@@ -64,6 +147,38 @@ where
     }
 }
 
+fn sub_opt<T>(lhs: Option<T>, rhs: Option<T>) -> Option<T>
+where
+    T: Sub<T, Output = T>,
+{
+    match (lhs, rhs) {
+        (Some(a), Some(b)) => Some(a - b),
+        (Some(a), None) => Some(a),
+        (None, Some(b)) => Some(b),
```

Comment:
Maybe Some(-b)  and T: Neg

Reply:
I'm not clear on why. Please expand.

Reply:
Also, these functions are structured per the existing `max_opt` and `min_opt` functions.